plugins/performance-lab/includes/site-health/far-future-headers/helper.php

```diff
@@ -0,0 +1,183 @@
+<?php
+/**
+ * Helper function to detect if static assets have far-future expiration headers.
+ *
+ * @package performance-lab
+ * @since n.e.x.t
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Callback for the far-future caching test.
+ *
+ * @since n.e.x.t
+ *
+ * @return array{label: string, status: string, badge: array{label: string, color: string}, description: string, actions: string, test: string} Result.
+ */
+function perflab_ffh_assets_test(): array {
+	$result = array(
+		'label'       => __( 'Your site serves static assets with far-future expiration headers', 'performance-lab' ),
+		'status'      => 'good',
+		'badge'       => array(
+			'label' => __( 'Performance', 'performance-lab' ),
+			'color' => 'blue',
+		),
+		'description' => sprintf(
+			'<p>%s</p>',
+			esc_html__(
+				'Serving static assets with far-future expiration headers improves performance by allowing browsers to cache files for a long time, reducing repeated requests.',
+				'performance-lab'
+			)
+		),
+		'actions'     => '',
+		'test'        => 'is_far_future_headers_enabled',
+	);
+
+	// List of assets to check.
+	$assets = array(
+		plugins_url( 'far-future-headers/assets/test.css', __DIR__ ),
+	);
+
+	// Check if far-future headers are enabled for all assets.
+	$status = perflab_ffh_check_assets( $assets );
+
+	if ( 'good' !== $status ) {
+		$result['status']  = $status;
+		$result['label']   = __( 'Your site does not serve static assets with recommended far-future expiration headers', 'performance-lab' );
+		$result['actions'] = sprintf(
+			'<p>%s</p>',
+			esc_html__( 'Far-future Cache-Control or Expires headers can be added or adjusted with a small configuration change by your hosting provider.', 'performance-lab' )
+		);
+	}
```

Comment:
There are two different checks which are getting performed one with `Cache-Control`, `Expires` and other with `Etag`, `Last-Modified` should there be different messages shown based on the checks?

Reply:
In https://github.com/WordPress/performance/pull/1727/commits/3db58860f1bbbd47a23824e05adb2aa8938d6125 table will be used to display reason for different failure cases.